### PR TITLE
NFSE-5054: nightly tests w/ smoke and stress

### PR DESCRIPTION
**Pull request by leroyjvargis**
## Description
Add nightly tests for github actions. Includes tests from the stress and smoke suites. 

## Verified checks?
Have the checks completed?
- [x] meson test -C build --setup=ci
- [x] All commits are signed off

### CODE REVIEW

**Review by alexttx — APPROVED** [review]

File: .github/workflows/nightly_build.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: Nightly builds
```
Comment (tristan957) [inline .github/workflows/nightly_build.yaml:1]: Call it "Nightly Testing" and rename the file to "nightly_testing.yaml"

File: .github/workflows/nightly_build.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        required: true
+        default: "master"
+
+jobs:
+  stress_ub:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04]
```
Comment (tristan957) [inline .github/workflows/nightly_build.yaml:17]: Use Ubuntu 20.04

File: tests/stress/meson.build [hunk 3]
```diff
@@ -136,8 +195,19 @@ foreach p, p_params: programs
             e,
             args: t_params['args'],
             is_parallel: false,
-            suite: ['stress_large'],
+            suite: ['stress-large'],
             timeout: params.get('timeout', 3600),
         )
     endforeach
+
+    foreach t, t_params : p_params['gh_tests']
```
Comment (tristan957) [inline tests/stress/meson.build:203]: any better name than gh_tests? Like "nightly-tests" maybe?

**Review by tristan957 — CHANGES_REQUESTED** [review]

**Comment by tristan957** [comment]
Make sure to pass `-s` to `git commit`.

File: .github/workflows/nightly_testing.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+          echo "Removing large directories"
+          # deleting 15GB
+          rm -rf /usr/share/dotnet/
+          df -h
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install build-essential libbsd-dev pkg-config openjdk-8-jdk libmicrohttpd-dev liburcu-dev libyaml-dev liblz4-dev libcurl4-openssl-dev libmongoc-1.0-0 libbson-1.0-0 libssl-dev libsasl2-dev
+          python3 -m pip install --user setuptools_rust wheel poetry
```
Comment (tristan957) [inline .github/workflows/nightly_testing.yaml:37]: Does install poetry not also install wheel and setuptools_rust?

Comment (leroyjvargis) [reply .github/workflows/nightly_testing.yaml:37]: it does, will update.

**Comment by tristan957** [comment]
@bhaveshvasandani I suggested to Leroy he use 20.04. Do you have any thoughts? Should we be running these tests nightly on Fedora as well?

**Review by tristan957 — APPROVED** [review]

**Pull request merged**